pkg/util/vralloc/alloc.go

```diff
@@ -97,15 +114,47 @@ func (a *FixedSizeAllocator[T]) Allocate(id T, r *Resource) (allocated bool, sho
 	return false, short
 }
 
-func (a *FixedSizeAllocator[T]) Release(id T) {
+func (a *FixedSizeAllocator[T]) Reallocate(id T, delta *Resource) (allocated bool, short *Resource) {
+	a.lock.Lock()
+	r, ok := a.allocs[id]
+	a.lock.Unlock()
+
+	if !ok {
+		return a.Allocate(id, delta)
+	}
+
+	a.lock.Lock()
+	defer a.lock.Unlock()
+	r.Add(delta)
+	if !zero.Le(r) {
```

Comment:
in which case r will less than 0

Reply:
When delta of reallocation is more than its holding, e.g., holding (10,10,10) but reallocate (-20,-20,-20), which is not allowed.